src/app/store/shopping-list/shopping-list.effects.ts

```diff
@@ -149,6 +147,32 @@ export class shoppingListEffects {
 		);
 	});
 
+	updateIngredientsFromRecipe = createEffect(() => {
+		return this.actions$.pipe(
+			ofType(shoppingListActions.updateIngredient),
+			concatLatestFrom(() =>
+				this.store.select(ShoppingListSelectors.selectAll)
+			),
+			map(([action, ingredientInShoppingList]) => {
+				let ingredientsToStore = ingredientInShoppingList;
+				const { ingredient, previewIngredient } = action;
+
+				ingredientsToStore = ingredientsToStore.filter(
+					ingredient => ingredient !== previewIngredient
+				);
```

Comment:
this code makes no sense because `ingredientsToStore` variable is overridden 2 lines below

Reply:
This piece of code remove  previous ingredient. After that I add new modified element. This protect from duplicate element on list. Then I add amount of ingredient with same name and unit. If I don't remove previous element , addition  have wrong value